packages/docs/src/routes/docs/cookbook/re-exporting-loaders/index.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: Cookbook | Re-exporting loaders
+contributors:
+  - gioboa
+updated_at: '2023-12-15T11:00:00Z'
+created_at: '2023-12-15T11:00:00Z'
+---
+
+# Re-exporting loaders
+
+`routeAction$` and `routeLoader$` are typically declared in route boundary files such as layout.tsx, index.tsx and plugin.tsx inside the routesDir directory
+[here is docs](https://qwik.builder.io/docs/route-loader/).
+
+Somethimes you would like to declared them outside of the route boundaries.
+This may be useful when you want to create reusable logic or a library.
+In such a case it is important that this function is re-exported from within the router boundary otherwise it will not run or throw exception.
+
+## Solution
+
+You can define `routeAction$` and `routeLoader$` in your custom path and re-export them in your layout.tsx, index.tsx and plugin.tsx files.
+
+Example with this path `./src/shared/loaders.ts`
+
+```tsx
+import { routeAction$, routeLoader$ } from '@builder.io/qwik-city';
+
+// eslint-disable-next-line qwik/loader-location
+export const useCommonRouteAction = routeAction$(async () => {
+  // ...
+});
+
+// eslint-disable-next-line qwik/loader-location
+export const useCommonRouteLoader = routeLoader$(async () => {
+  // ...
+});
+```
```

Comment:
I am always concerned tha these code examples will go out of date ar will have syntax error. Can we turn them into real code? See any `MDX` file which contains

```
import CodeSandbox, {CodeFile} from '../../../../components/code-sandbox/index.tsx';
```

Reply:
✅ done